Added JUnit5 tests for all classes

Januzao opened the pull request.

Januzao committed:
Added JUnit5 tests for all classes

Elena-Bruyako commented on src/test/java/core/basesyntax/service/impl/DataConverterImplTest.java:
```diff
@@ -0,0 +1,113 @@
+        List<String> lines = List.of("b banana 20");
+
+        List<FruitTransaction> fruitTransactions = dataConverter.convertToTransaction(lines);
+        Assertions.assertEquals(1, fruitTransactions.size());
```
use static imports
check and fix all places

Elena-Bruyako commented on src/test/java/core/basesyntax/service/impl/ReadCsvFileImplTest.java:
```diff
@@ -0,0 +1,58 @@
+class ReadCsvFileImplTest {
+    @Test
+    void readCsvFile_validFile_ok() {
+        ReadCsvFileImpl readCsvFile = new ReadCsvFileImpl();
```
initialise in `@BeforeEach `method
check and fix all places

Elena-Bruyako commented on src/test/java/core/basesyntax/service/impl/ReportGeneratorImplTest.java:
```diff
@@ -0,0 +1,58 @@
+class ReportGeneratorImplTest {
+    @Test
+    void constructor_nullStorage_throwsNullPointerException() {
+        Assertions.assertThrows(NullPointerException.class,
```
don't test NullPointerException
you should avoid this exception in main code

Elena-Bruyako reviewed (changes requested):
clear Storage in `@AfterEach` method in all tests
fix other comments

Januzao committed:
Refactor: Improve test structure and fix code style

Januzao reviewed (commented):
Refactored test classes to use "BeforeEach" for object initialization and "AfterEach" to clear Storage state, ensuring better test isolation.
Added a `clear()` method to the Storage interface and implementation.
Switched to static imports.
Removed tests that explicitly check for `NullPointerException` as they should be avoided in the main code.

sarakhmen commented on src/test/java/core/basesyntax/service/impl/DataConverterImplTest.java:
```diff
@@ -0,0 +1,113 @@
+        assertEquals(FruitTransaction.Operation.BALANCE,
+                fruitTransaction.getOperation());
+        assertEquals("banana", fruitTransaction.getFruit());
+        assertEquals(20, fruitTransaction.getQuantity());
```
I'd recommend comparing the whole Actual result with the full expected data. It could be done in 1-2 lines of code maximum, instead of comparing each field separately. Fix the same issue in all places

sarakhmen commented on src/test/java/core/basesyntax/db/impl/StorageImplTest.java:
```diff
@@ -0,0 +1,82 @@
+class StorageImplTest {
+    private Storage storage;
+
+    @BeforeEach
```
actually tests for StorageImpl could be removed, it's redundant

Januzao replied:
Hey, could you clarify why StorageImplTest.java should be removed? I thought testing the defensive copy logic was important here.

sarakhmen replied:
your storage impl is essentialy a simple wrapper around hashmap. It's not a service or any kind of business logic that should be tested here

Januzao replied:
okay, got it

sarakhmen commented on src/test/java/core/basesyntax/service/impl/ReadCsvFileImplTest.java:
```diff
@@ -0,0 +1,61 @@
+
+    @Test
+    void readCsvFile_validFile_ok() {
+        List<String> lines = readCsvFile.readCsvFile("fruits.csv");
```
I don't see your fruits.csv file in test resources folder

sarakhmen replied:
same for other files

sarakhmen replied:
please move test data from main resources to test resources folder

sarakhmen commented on pom.xml:
```diff
@@ -32,6 +33,12 @@
+            <artifactId>mockito-core</artifactId>
+            <version>${mockito.version}</version>
+            <scope>test</scope>
+        </dependency>
```
I don't see you using mockito for your tests. Please remove this dependency

sarakhmen commented on pom.xml:
```diff
@@ -17,6 +17,7 @@
         </maven.checkstyle.plugin.configLocation>
         <jacoco.version>0.8.10</jacoco.version>
         <junit.jupiter.version>5.9.3</junit.jupiter.version>
+        <mockito.version>5.3.1</mockito.version>
```
and this mockito property as well

sarakhmen reviewed (changes requested):
let's improve your solution :)

Januzao committed:
Refactor: Improve test structure
Implemented equals/hashCode in FruitTransaction for direct object comparison.
Refactored tests to compare full collections/objects instead of individual fields.
Moved test files to test/resources directory
Removed StorageImplTest.java as redundant
Removed unused dependencies

sarakhmen reviewed (approved):
lgtm!
